app/services/partner_invite_and_approval_service.rb

```diff
@@ -0,0 +1,44 @@
+class PartnerInviteAndApprovalService
+  include ServiceObjectErrorsMixin
+
+  def initialize(partner:)
+    @partner = partner
+  end
+
+  def call
+    existing_user = User.find_by(email: partner.email)
+    if existing_user && existing_user.partner_id.nil?
+      existing_user.update!(partner_id: partner.profile.id)
+      return self
+    end
+
+    partner.update!(status: 'invited')
+    UserInviteService.invite(email: partner.email,
+      roles: [Role::PARTNER],
+      resource: partner)
```

Comment:
Aren't these lines all taken from the PartnerInviteService? Is there a reason you can't call the PartnerInviteService directly rather than copying the lines of code from there?